From 66b71951f8ce66f02040f8814e1d89db34c34a87 Mon Sep 17 00:00:00 2001
From: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Date: Sun, 22 Feb 2015 12:36:00 +0100
Subject: [PATCH] os-compatibility: define CLOCK_BOOTTIME when not available

CLOCK_BOOTTIME is supposed to be provided by the C library headers,
but uClibc 0.9.33.2 does not provide it. Instead of depending on an
uClibc patch, let's make Weston define it to the right value when not
already defined by the C library.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
---
 shared/os-compatibility.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/shared/os-compatibility.h b/shared/os-compatibility.h
index 172bb7e..e369899 100644
--- a/shared/os-compatibility.h
+++ b/shared/os-compatibility.h
@@ -26,6 +26,7 @@
 #include "config.h"
 
 #include <sys/types.h>
+#include <time.h>
 
 #ifdef HAVE_EXECINFO_H
 #include <execinfo.h>
@@ -37,6 +38,10 @@ backtrace(void **buffer, int size)
 }
 #endif
 
+#ifndef CLOCK_BOOTTIME
+#define CLOCK_BOOTTIME 7
+#endif
+
 int
 os_socketpair_cloexec(int domain, int type, int protocol, int *sv);
 
-- 
2.1.0

